src/main/java/ogc/rs/health/HealthCheckService.java

```diff
@@ -0,0 +1,676 @@
+        String latestFileVersion = getLatestMigrationVersionFromFiles();
+
+        // Check database migration status
+        String query = "SELECT version, success FROM flyway_schema_history ORDER BY installed_rank DESC LIMIT 1";
+
+        pgPool.query(query).execute(ar -> {
+            JsonObject result = new JsonObject().put("name", "flyway-migration-status");
+
+            if (ar.succeeded()) {
+                RowSet<Row> rows = ar.result();
+                if (rows.size() > 0) {
+                    Row row = rows.iterator().next();
+                    String dbVersion = row.getString("version");
+                    Boolean success = row.getBoolean("success");
+
+                    boolean isUpToDate = latestFileVersion != null && latestFileVersion.equals(dbVersion);
+                    boolean isSuccessful = success != null && success;
+
+                    if (isUpToDate && isSuccessful) {
+                        result.put("status", STATUS_UP)
+                                .put("data", new JsonObject()
+                                        .put("latestFileVersion", latestFileVersion)
+                                        .put("databaseVersion", dbVersion)
+                                        .put("migrationSuccess", success)
+                                        .put("upToDate", true));
+                    } else {
+                        result.put("status", STATUS_DOWN)
+                                .put("data", new JsonObject()
+                                        .put("latestFileVersion", latestFileVersion)
+                                        .put("databaseVersion", dbVersion)
+                                        .put("migrationSuccess", success)
+                                        .put("upToDate", false));
+                    }
+                } else {
+                    result.put("status", STATUS_DOWN)
+                            .put("data", new JsonObject()
+                                    .put("error", "No migration history found"));
+                }
+            } else {
+                result.put("status", STATUS_DOWN)
+                        .put("data", new JsonObject()
+                                .put("error", ar.cause().getMessage()));
+            }
+
+            promise.complete(result);
+        });
+
+        return promise.future();
+    }
+
+    /**
+     * Checks catalogue service availability by making HTTP request to configured endpoint.
+     * Accepts 2xx and 4xx status codes as indication that service is running.
+     */
+    private Future<JsonObject> checkCatalogueService() {
+        Promise<JsonObject> promise = Promise.promise();
+
+        String catServerHost = config.getString("catServerHost");
+        Integer catServerPort = config.getInteger("catServerPort");
+        String catRequestInstanceUri = config.getString("catRequestInstanceUri", "/iudx/cat/v1/internal/ui/instance");
```

Comment:
It may be better to use the normal cat APIs like list or search instead of the internal ones as there is no guarantee they will remain.